crypto/s2n_ecc_evp.h

```diff
@@ -34,21 +34,21 @@ struct s2n_ecc_named_curve {
 
 extern const struct s2n_ecc_named_curve s2n_ecc_curve_secp256r1;
 extern const struct s2n_ecc_named_curve s2n_ecc_curve_secp384r1;
-
-#define S2N_ECC_EVP_SUPPORTED_CURVES_COUNT 2
+extern const struct s2n_ecc_named_curve s2n_ecc_curve_x25519;
 
 /* BoringSSL only supports using EVP_PKEY_X25519 with "modern" EC EVP APIs. BoringSSL has a note to possibly add this in
  * the future. See https://github.com/google/boringssl/blob/master/crypto/evp/p_x25519_asn1.c#L233
  */
 #if S2N_OPENSSL_VERSION_AT_LEAST(1, 1, 0) && !defined(LIBRESSL_VERSION_NUMBER) && !defined(OPENSSL_IS_BORINGSSL)
-#define MODERN_EC_SUPPORTED 1
-extern const struct s2n_ecc_named_curve s2n_ecc_curve_x25519;
+    #define MODERN_EC_SUPPORTED 1
+    #define S2N_ECC_EVP_SUPPORTED_CURVES_COUNT 3
```

Comment:
this might be less sustainable in future when there are more combinations of libcrypto + curves. one approach would be to align with how `cipher_suite->available` with the curves, but we can probably live with this for the time